src/ds/ds-motion-common.cpp

```diff
@@ -0,0 +1,525 @@
+
+        if (stream == RS2_STREAM_GYRO)
+            return ds::create_motion_intrinsics(**_gyro_intrinsic);
+
+        throw std::runtime_error(rsutils::string::from() << "Motion Intrinsics unknown for stream " << rs2_stream_to_string(stream) << "!");
+    }
+
+    std::vector<platform::uvc_device_info> ds_motion_common::filter_device_by_capability(const std::vector<platform::uvc_device_info>& devices,
+        d400_caps caps)
+    {
+        if (auto dev = dynamic_cast<const ds5_motion*>(_owner))
+            return filter_ds5_device_by_capability(devices, ds::d400_caps::CAP_FISHEYE_SENSOR);
+        if (auto dev = dynamic_cast<const ds5_motion_uvc*>(_owner))
+            return filter_ds5_device_by_capability(devices, ds::d400_caps::CAP_FISHEYE_SENSOR);
+        throw std::runtime_error("device not referenced in the product line");
+    }
+
+    std::vector<platform::uvc_device_info> ds_motion_common::init_fisheye(const platform::backend_device_group& group, bool& is_fisheye_available)
+    {
+        auto fisheye_infos = filter_by_mi(group.uvc_devices, 3);
+        fisheye_infos = filter_device_by_capability(fisheye_infos, ds::d400_caps::CAP_FISHEYE_SENSOR);
+
+        bool fe_dev_present = (fisheye_infos.size() == 1);
+        bool fe_capability = (ds::d400_caps::CAP_UNDEFINED == _device_capabilities) ?
+            true : !!(static_cast<uint32_t>(_device_capabilities & ds::d400_caps::CAP_FISHEYE_SENSOR));
+
+        // Motion module w/o FishEye sensor
+        if (!(fe_dev_present | fe_capability))
+            return fisheye_infos;
+
+        // Inconsistent FW
+        if (fe_dev_present ^ fe_capability)
+            throw invalid_value_exception(rsutils::string::from()
+                << "Inconsistent HW/FW setup, FW FishEye capability = " << fe_capability
+                << ", FishEye devices " << std::dec << fisheye_infos.size()
+                << " while expecting " << fe_capability);
+        
+        is_fisheye_available = true;
+
+        _fisheye_calibration_table_raw = [this]()
+        {
+            return _mm_calib->get_fisheye_calib_raw();
+        };
+
+        return fisheye_infos;
+    }
+
+    void ds_motion_common::assign_fisheye_ep(std::shared_ptr<uvc_sensor> raw_fisheye_ep,
+        std::shared_ptr<synthetic_sensor> fisheye_ep,
+        std::shared_ptr<global_time_option> enable_global_time_option)
+    {
+        _raw_fisheye_ep = raw_fisheye_ep;
+        _fisheye_ep = fisheye_ep;
+        _enable_global_time_option = enable_global_time_option;
+    }
+
+    void ds_motion_common::set_roi_method()
+    {
+        auto fisheye_auto_exposure = register_auto_exposure_options(_fisheye_ep.get(), &ds::fisheye_xu);
+        if (auto fisheye_sensor = dynamic_cast<ds_fisheye_sensor*>(_fisheye_ep.get()))
+            fisheye_sensor->set_roi_method(std::make_shared<fisheye_auto_exposure_roi_method>(fisheye_auto_exposure));
+        else if (auto fisheye_sensor = dynamic_cast<ds_fisheye_sensor*>(_fisheye_ep.get()))
```

Comment:
if and else match,
Am I missing something?

Reply:
correct- removed